test/chainrules.jl

```diff
@@ -57,14 +57,10 @@
         test_rrule(logcosh, x)
     end
 
-    # test all branches of `log1pexp`
-    for x in (-20.9, 15.4, 41.5)
-        test_frule(log1pexp, x)
-        test_rrule(log1pexp, x)
-    end
-    for x in (8.3f0, 12.5f0, 21.2f0)
-        test_frule(log1pexp, x; rtol=1f-3, atol=1f-3)
-        test_rrule(log1pexp, x; rtol=1f-3, atol=1f-3)
+    @testset "log1pexp on $T" for T in (Float64, BigFloat)
```

Comment:
Why did you not check Float32 as well? The main "issue" there should be that one has to adjust the tolerances, as in the current tests, due to issues/inaccuracies of FiniteDifferences with Float32.

Reply:
Ok I adjusted the tols and now I am checking Float32 too. Thanks!